runtime/lua/vim/lsp/util.lua

```diff
@@ -90,6 +90,44 @@ local function split_lines(value)
   return split(value, '\n', true)
 end
 
+---Alternative to vim.str_utfindex that takes an encoding.
+---@param line string line to be indexed
+---@param index number byte index (utf-8)
+---@param encoding string utf-8|utf-16|utf-32|nil defaults to utf-16
+---@return number `encoding` index of `index` in `line`
+function M.str_utfindex(line, index, encoding)
+  if encoding ~= protocol.Encodings.UTF8 then
+    local col32, col16
+    if index then
+      col32, col16 = vim.str_utfindex(line, index)
+    else
+      col32, col16 = vim.str_utfindex(line)
+    end
+    if encoding == protocol.Encodings.UTF32 then
+      return col32
+    else
+      return col16
+    end
+  else
+    return index
+  end
+end
+
+---Alternative to vim.str_byteindex that takes an encoding.
+---@param line string line to be indexed
+---@param index number UTF index
+---@param encoding string utf-8|utf-16|utf-32|nil defaults to utf-16
+---@return number byte (utf-8) index of `encoding` index `index` in `line`
+function M.str_byteindex(line, index, encoding)
+  if encoding ~= protocol.Encodings.UTF8 then
+    return vim.str_byteindex(line, index, not encoding or encoding ~= protocol.Encodings.UTF32)
+  else
+    return index
+  end
+end
+
+local str_utfindex, str_byteindex = M.str_utfindex, M.str_byteindex
```

Comment:
We already have `character_offset` and `get_line_byte_from_position` which are lsp specific.
If we count the `vim.` functions we'd have a total of 6 functions for the conversions. Do we really need all of these? 

The overlapping naming makes it also kinda confusing and easy to mix them up

Reply:
I don't think any existing function does what we're doing here, but I'm happy to rename it, did you have anything in mind?